core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -582,5 +585,28 @@ class CityInfo {
         for(otherCiv in civsWithCloseCities)
             otherCiv.getDiplomacyManager(civInfo).setFlag(DiplomacyFlags.SettledCitiesNearUs,30)
     }
+
+    private fun hasUnitOfType(type: UnitType) : Boolean
+    {
+        for(unit in getCenterTile().getUnits()) {
```

Comment:
This can be simplified to return getUnits().any{it.type==type}

Reply:
very true!